core/domain/question_services.py

```diff
@@ -138,7 +138,10 @@ def get_questions_and_skill_descriptions_by_skill_ids(
     question_skill_links = [
         get_question_skill_link_from_model(model, skill_descriptions[index])
         for index, model in enumerate(question_skill_link_models)]
-    question_ids = [obj.question_id for obj in question_skill_links]
+    question_ids = []
+    for obj in question_skill_links:
+        if obj.question_id not in question_ids:
```

Comment:
This can be done more efficiently using python sets and `question_ids.add()` in that. Maybe try that?
Then, you can  convert to list later for query purposes.